src/test/java/com/google/sps/api/GmailConfigurationTest.java

```diff
@@ -0,0 +1,163 @@
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalURLFetchServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.api.client.util.Base64;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.model.Message;
+import com.google.sps.data.Email;
+import com.google.sps.data.Post;
+import com.google.sps.servlets.PostDataServlet;
+import com.google.sps.api.GmailConfiguration;
+import com.google.sps.api.GmailAPI;
+import com.google.sps.MemoryAppender;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Properties;
+import javax.mail.internet.MimeMessage;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.LoggerFactory;
+
+/** Tests querying users and sending emails from an authorized account.*/
+@RunWith(JUnit4.class)
+public final class GmailConfigurationTest {
+
+  private static final String TO = "test@google.com";
+  private static final String SUBJECT = "test";
+  private static final String CONTENT = "<h1>test</h1>";
+
+  private static final String NAME = "test";
+  private static final String EMAIL = "test@google.com";
+  private static final String COLLEGE_A = "000000";
+  private static final String COLLEGE_B = "000001";
+
+  private static final String SUCCESS_MSG = "Successfully sent a new post email to: ";
+  private static final String LOGGER_NAME = "com.google.sps.api";
+
+  @Mock private static Gmail mGmail;
+  @Mock private static MimeMessage mMimeMessage;
+  @Mock private static GmailAPI mGmailAPI;
+  @Mock private static PreparedQuery mpq;
+
+  private static LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private static DatastoreService datastore;
+  private static GmailConfiguration gmailConfiguration;
+  private static MemoryAppender memoryAppender;
+  private static Post post;
+
+  @Before
+  public void setUp() throws Exception {
+
+    helper.setUp();
+
+    Logger logger = (Logger) LoggerFactory.getLogger(LOGGER_NAME);
+    memoryAppender = new MemoryAppender();
+    memoryAppender.setContext((LoggerContext) LoggerFactory.getILoggerFactory());
+    logger.setLevel(Level.DEBUG);
+    logger.addAppender(memoryAppender);
+    memoryAppender.start();
+
+    post = new Post();
+    datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
is this a mocked datastore? :0

Reply:
Nope! For the `userAttendingCollege()`, it has to be a real datastore to query through and get results as that's what we will return!

Reply:
Doesn't this violate the principle of not relying on external dependencies in unit tests? Or are these integration tests?

Reply:
For Datastore testing, I followed Google's standards: https://cloud.google.com/appengine/docs/standard/java/tools/localunittesting!

Reply:
Ah I see! It's a local instance of datastore and so it doesn't have to make any network calls :) makes sense!